docs/question/template.md

```diff
@@ -1,6 +1,116 @@
+
+### `"submission"` panel
+
+By default, feedback given by elements (`data["partial_scores"][NAME]["feedback"]`) is rendered when an element renders in the "submission" panel. You can use the [`pl-submission-panel`](../elements.md#pl-submission-panel-element) to show additional text only in the "submission" panel. For example:
+
+=== "Good"
+
+    ```html
+    <pl-submission-panel>
+      <p>
+        Feedback: {{feedback.y}}
+      </p>
+    </pl-submission-panel>
+    ```
+
+=== "Bad"
+
+    ```html
+    <p>Feedback: {{feedback.y}}</p>
+    ```
+
+    This will show the feedback in the "question" and "answer" panels as well, which is unnecessary and can be confusing for students. Since `feedback.y` won't be defined until the student submits an answer, it will show as an empty string in the "question" panel, just showing `"Feedback: "`.
+
+More details about providing feedback can be found on the [`server.py` page](server.md#providing-feedback).
+
+### `"answer"` panel
+
+By default, the correct answers are shown in the "answer" panel. This can be disabled by setting `showCorrectAnswer: false` in the question's `info.json`. The FAQ item ["How can I hide the correct answer when students see their grading results?"](../faq.md#how-can-i-hide-the-correct-answer-when-students-see-their-grading-results) describes some other techniques for hiding the correct answer.
+
+If you grade a question using a [custom grading function](./server.md#step-5-grade) or an [external grader](../externalGrading.md), it is recommended to set the correct answer so that students can see it in the answer panel. If there are multiple correct answers, you should add a note within the [`pl-answer-panel`](../elements.md#pl-answer-panel-element) element that any correct answer would be accepted, and the displayed answer is only an example. You can also show additional text in the answer panel as well, e.g. extended explanations for the correct answers.
+
+## Mustache templates
+
+Any text in double-curly-braces (like `{{params.m}}`) is substituted with variable values using [Mustache](https://mustache.github.io/mustache.5.html). These parameters are typically defined by a [question's `server.py`](server.md). For example, if `params["m"]` is defined in `server.py`, you can use it in `question.html` like this:
+
+```html title="question.html"
+<p>If m is {{params.m}}, what is {{params.m}} * 3?</p>
+```
+
+This will be rendered as:
+
+```html
+<p>If m is 5, what is 5 * 3?</p>
+```
+
+### Advanced rendering
+
+You can use the following syntax to conditionally render a piece of HTML:
+
+```html title="question.html"
+{{#params.show}}
+<p>This will only show up if params.show is true</p>
+{{/params.show}}
+```
+
+You can use this syntax to render over an array of strings (the `.` represents the current item in the array):
+
+```html title="question.html"
+{{#params.items}}
+<p>{{.}}</p>
+{{/params.items}}
+```
+
+### HTML rendering
```

Comment:
This is a single paragraph, does it need a heading?

Reply:
Changed to a tip.

Reply:
Changed in 6408ede3ac9fe51ab8d193ea476887f9d181752a.